internal/page/donor/your_name.go

```diff
@@ -0,0 +1,131 @@
+package donor
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/gorilla/sessions"
+	"github.com/ministryofjustice/opg-go-common/template"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/actor"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/page"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/sesh"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/validation"
+)
+
+type yourNameData struct {
+	App         page.AppData
+	Errors      validation.List
+	Form        *yourNameForm
+	NameWarning *actor.SameNameWarning
+}
+
+func YourName(tmpl template.Template, donorStore DonorStore, sessionStore sessions.Store) Handler {
+	return func(appData page.AppData, w http.ResponseWriter, r *http.Request, donor *actor.DonorProvidedDetails) error {
+		data := &yourNameData{
+			App: appData,
+			Form: &yourNameForm{
+				FirstNames: donor.Donor.FirstNames,
+				LastName:   donor.Donor.LastName,
+				OtherNames: donor.Donor.OtherNames,
+			},
+		}
+
+		if r.Method == http.MethodGet && data.Form.FirstNames == "" {
+			if latestDonor, _ := donorStore.Latest(r.Context()); latestDonor != nil {
+				data.Form = &yourNameForm{
+					FirstNames: latestDonor.Donor.FirstNames,
+					LastName:   latestDonor.Donor.LastName,
+					OtherNames: latestDonor.Donor.OtherNames,
+				}
+			}
+		}
+
+		if r.Method == http.MethodPost {
+			loginSession, err := sesh.Login(sessionStore, r)
+			if err != nil {
+				return err
+			}
+			if loginSession.Email == "" {
+				return fmt.Errorf("no email in login session")
+			}
```

Comment:
could this be left out here, since you'll pick it up when you Continue and go through /your-details?